server/models/Collective.js

```diff
@@ -2513,6 +2513,24 @@ export default function (Sequelize, DataTypes) {
     }).then(result => Promise.resolve(parseInt(result.toJSON().total, 10)));
   };
 
+  Collective.prototype.getTotalNetAmountReceived = function (startDate, endDate) {
+    endDate = endDate || new Date();
+    const where = {
+      netAmountInCollectiveCurrency: { [Op.gt]: 0 },
+      createdAt: { [Op.lt]: endDate },
+      CollectiveId: this.id,
+    };
+    if (startDate) {
+      where.createdAt[Op.gte] = startDate;
+    }
+    return models.Transaction.findOne({
+      attributes: [
+        [Sequelize.fn('COALESCE', Sequelize.fn('SUM', Sequelize.col('netAmountInCollectiveCurrency')), 0), 'total'],
+      ],
+      where,
+    }).then(result => Promise.resolve(parseInt(result.toJSON().total, 10)));
```

Comment:
It's not new code but I'm surprised by this `.toJSON()`, it doesn't seem efficient. `Promise.resolve` also does not seem necessary.

Reply:
Agree. Will keep it like this for now.